raiden/tests/integration/rpc/test_assumptions.py

```diff
@@ -93,18 +113,18 @@ def test_estimate_gas_fail(deploy_client, blockchain_backend):
     assert not contract_proxy.estimate_gas('fail')
 
 
-def test_duplicated_transaction_raises(deploy_client, blockchain_backend):
+def test_duplicated_transaction_raises(deploy_client):
     """ If the same transaction is sent twice a JSON RPC error is raised. """
     contract_proxy = deploy_rpc_test_contract(deploy_client)
 
     address = contract_proxy.contract_address
     assert len(deploy_client.web3.eth.getCode(to_checksum_address(address))) > 0
 
-    host = '0.0.0.0'  # hardcoded in the deploy_client fixture
     second_client = JSONRPCClient(
-        host,
+        '',
```

Comment:
why did you remove the `'0.0.0.0'` host? Is empty string the same? If yes why remove it and not just be explicit?

Reply:
`host` and `port` are not used if `web3` is provided. I was thinking of removing those two variables completely and make `web3` a required argument, but I thought I'd address this in a separate PR as it also affects other parts of the code.

Reply:
Hmm I am not following.

Web3 is initialized in the ctor of `JSONRPCClient`:

```
        endpoint = 'http://{}:{}'.format(host, port)
        # web3
        self.web3: Web3 = Web3(HTTPProvider(endpoint))
```

Reply:
you are probably looking at a wrong branch:

https://github.com/pcppcp/raiden/blob/tester_test/raiden/network/rpc/client.py#L231

Reply:
Ah okay indeed. Now I see. So you want to make passing the web3 argument mandatory and then get rid of these args. Cool. This can happen in another PR ... but still in the meantime makes no sense to remove `host` from here in this PR.